tests/Support/NavigationTest.php

```diff
@@ -94,7 +96,7 @@ public static function providePeriodsWithSkippingParam(): array
             '2023-05-31 to 2023-07-31' => ['skip' => 1, 'frequency' => 'monthly', 'from' => Carbon::parse('2023-05-31'), 'expected' => Carbon::parse('2023-07-31')],
             '2023-08-31 to 2023-10-31' => ['skip' => 1, 'frequency' => 'monthly', 'from' => Carbon::parse('2023-08-31'), 'expected' => Carbon::parse('2023-10-31')],
             '2023-10-31 to 2023-12-31' => ['skip' => 1, 'frequency' => 'monthly', 'from' => Carbon::parse('2023-10-31'), 'expected' => Carbon::parse('2023-12-31')],
-            '2023-01-31 to 2023-03-30' => ['skip' => 2, 'frequency' => 'monthly', 'from' => Carbon::parse('2023-01-31'), 'expected' => Carbon::parse('2023-05-1')],
+            '2023-01-31 to 2023-03-30' => ['skip' => 2, 'frequency' => 'monthly', 'from' => Carbon::parse('2023-01-31'), 'expected' => Carbon::parse('2023-04-30')],
```

Comment:
@JC5, see [here](https://github.com/firefly-iii/firefly-iii/pull/7685/commits/ac53eb7cd784f722714788f845a51be24fb10e94#diff-a561aad172d8a865422ff210ed6b0b58b3cc11f4b6e4c4458c5c4afa54ca1949L99-L110). I found a bug in the original implementation. It was disregarded in the subtraction if the date difference was more than two months.